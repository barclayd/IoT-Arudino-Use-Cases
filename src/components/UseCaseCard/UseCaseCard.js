import React, {Component} from 'react';
import { Divider, Button } from 'antd';
import styles from './UseCaseCard.module.scss'

class UseCaseCard extends Component {
    render() {
        return (
            <div className={styles.UseCaseCard}>
                <h3 className={styles.Title}>Thermometer Case</h3>
                <Divider type="horizontal" />
                <p>Improved own provided blessing may peculiar domestic. Sight house has sex never. No visited raising gravity outward s</p>
                {/*<div>*/}
                    <Button type="primary" block className={styles.Button}>Start</Button>
                {/*</div>*/}
            </div>
        )
    }
}

export default UseCaseCard;