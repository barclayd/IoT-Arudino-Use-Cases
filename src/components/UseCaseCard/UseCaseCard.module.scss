@import "../../styles/base/settings";


.Button {
  margin-bottom: $s-size;
}

.UseCaseCard {
  border: 1px solid $light-blue;
  margin: $s-size;
  padding: 0 $m-size;
}

.Logo
.Title {
  font-weight: 300;
  margin: 0;
  text-align: center;
}