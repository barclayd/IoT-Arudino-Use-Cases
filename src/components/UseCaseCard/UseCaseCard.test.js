import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Card } from 'antd';


import ReactRouterEnzymeContext from 'react-router-enzyme-context';


import UseCaseCard from './UseCaseCard';
// import SensorLive from '../../components/LiveSensor/LiveSensor';


configure({adapter: new Adapter()});

const { Meta } = Card;
const mockedUseCase = {
    "access" : {
      "community" : false,
      "listedUsers" : [ "kwLX0IevWZMyQmhESLU4CMGj2gO2", "l5AExuLCwlOwevsTCBI8f27q3Vt1", "qB0dbGthCDZwRhcNVRdAnslEXES2", "WcgsnhsRsiSxRykCmrnE2Vv4Pwe2", "MYPw0PQJAzb9XhnWlpMIY837yQ43", "tddO4URF3Qh6mxJoBTIPhkyR0Qb2" ],
      "student" : true,
      "trainer" : true
    },
    "arduino" : "tmp36.jpg",
    "arduinoDesc" : "The LM35 series are precision integrated-circuit temperature devices with an output voltage linearly-proportional to the Centigrade temperature. The LM35 device has an advantage over linear temperature sensors calibrated in Kelvin, as the user is not required to subtract a large constant voltage from the output to obtain convenient Centigrade scaling. The LM35 device does not require any external calibration or trimming to provide typical accuracies of ±¼°C at room temperature and ±¾°C over a full −55°C to 150°C temperature range.",
    "arduinoName" : "LM35",
    "email" : {
      "body" : "Fridge door has been left open 🌋🖍",
      "senders" : [ [ "t@t.com" ] ],
      "subject" : "ALERT ⚠️✂️ FRIDGE DOOR IS OPEN 🚨😩⚠️🙄"
    },
    "id" : "0",
    "image" : "nhsfridge.jpg",
    "imageDesc" : "A fridge image",
    "key" : "nhsfridgetemp",
    "longDesc" : "If you leave the fridge door open, heat is recycled from the room into the refrigerator, then back into the room. A net room temperature increase would result from the heat of the motor that would be constantly running to move energy around in a circle. This makes precious, double-blind research drugs at great risk of becoming spoilt and along with other medical equipment, all of which are vitally important for application to patients within hospitals whose lives depend on the application of working drugs and medicines.",
    "messages" : [ {
      "authorName" : "Peter Trott",
      "date" : "5th Dec 2018 | 17:04 PM",
      "message" : "Please update sensor values and components",
      "profileImage" : "beach.jpg",
      "rating" : 2.5,
      "title" : "Needs improvement"
    }, {
      "authorName" : "Peter Trott",
      "date" : "9th Dec 2018 | 18:04 PM",
      "message" : "Very good work on this Use Case - I am very impressed",
      "profileImage" : "nhsfridge.jpg",
      "rating" : 4,
      "title" : "Top Work"
    } ],
    "name" : "NHS Fridges 💉",
    "sensorsData" : [ {
      "maxValue" : 10,
      "minValue" : 7,
      "sensorComponent" : "LM35",
      "sensorName" : "Temperature Sensor"
    } ],
    "shortDesc" : "Check the temperature of fridges in University Hospital, Cardiff"
  }

  describe('<UseCard />', () => {
    let wrapper
    // before if test "it()" do the below
    beforeEach(() => {
        wrapper = shallow(<UseCaseCard isLoading={false} {...mockedUseCase} />);
    })

    it('should renders correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('component should render the correct useCaseCard values', () => {
        const card = wrapper.find('.card');
        expect(card.find(Meta).prop('title')).toEqual(mockedUseCase.name);
        expect(card.find(Meta).prop('description')).toEqual(mockedUseCase.shortDesc);
    })
})