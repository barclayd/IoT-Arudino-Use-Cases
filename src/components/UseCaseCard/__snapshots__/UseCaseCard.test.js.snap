// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseCard /> should renders correctly 1`] = `
<div
  className="UseCaseCard"
>
  <Card
    actions={
      Array [
        <Icon
          aria-label="experiment icon"
          style={
            Object {
              "fontSize": "22px",
            }
          }
          theme="filled"
          type="profile"
        />,
      ]
    }
    aria-label="Use case card, Use case name is NHS Fridges 游눌, Use case description is Check the temperature of fridges in University Hospital, Cardiff"
    bodyStyle={
      Object {
        "minHeight": 150,
      }
    }
    bordered={true}
    className="card"
    cover={
      <img
        alt="NHS Fridges 游눌"
        height="250px"
        src="/images/nhsfridge.jpg"
      />
    }
    hoverable={true}
    style={
      Object {
        "width": 300,
      }
    }
  >
    <div
      aria-label="Use case name: NHS Fridges 游눌, Use case description: Check the temperature of fridges in University Hospital, Cardiff"
      className="content"
    >
      <_default
        aria-label="Use case name: NHS Fridges 游눌, Use case description: Check the temperature of fridges in University Hospital, Cardiff"
        description="Check the temperature of fridges in University Hospital, Cardiff"
        style={
          Object {
            "textAlign": "center",
          }
        }
        title="NHS Fridges 游눌"
      />
    </div>
  </Card>
</div>
`;
